src/components/AddComment.jsx

```diff
@@ -0,0 +1,63 @@
+import { useContext, useState } from "react";
+import { postComment } from "../utils/requests";
+import UserContext from "../contexts/UserContext";
+import CommentCard from "./CommentCard";
+import "../styles/AddComment.css";
+
+const AddComment = ({ article_id }) => {
+  const { activeUser } = useContext(UserContext);
+  const [isCommenting, setIsCommenting] = useState(false);
+  const [newComment, setNewComment] = useState("");
+  const [postedComment, setPostedComment] = useState(null);
+  const [hasPosted, setHasPosted] = useState(false);
```

Comment:
Could you derive the value of the `hasPosted` state from the `postedComment` state?